com.unity.perception/Runtime/Randomization/Scenarios/ScenarioBase.cs

```diff
@@ -149,40 +189,182 @@ void Update()
+                throw new ScenarioException(
+                    $"Cannot add non-randomizer type {randomizerType.Name} to randomizer list");
+            foreach (var randomizer in m_Randomizers)
+                if (randomizer.GetType() == randomizerType)
+                    throw new ScenarioException(
+                        $"Two Randomizers of the same type ({randomizerType.Name}) cannot both be active simultaneously");
+            var newRandomizer = (Randomizer)Activator.CreateInstance(randomizerType);
+            m_Randomizers.Add(newRandomizer);
+            newRandomizer.m_Scenario = this;
+            newRandomizer.m_TagManager = tagManager;
+            newRandomizer.Create();
+            return newRandomizer;
+        }
+
+        /// <summary>
+        /// Removes a randomizer of the specified type from this scenario
+        /// </summary>
+        /// <typeparam name="T">The type of scenario to remove</typeparam>
+        public void RemoveRandomizer<T>() where T : Randomizer, new()
+        {
+            RemoveRandomizer(typeof(T));
+        }
+
+        internal void RemoveRandomizer(Type randomizerType)
+        {
+            if (!randomizerType.IsSubclassOf(typeof(Randomizer)))
+                throw new ScenarioException(
+                    $"Cannot add non-randomizer type {randomizerType.Name} to randomizer list");
+            var removed = false;
+            for (var i = 0; i < m_Randomizers.Count; i++)
+            {
+                if (m_Randomizers[i].GetType() == randomizerType)
+                {
+                    m_Randomizers.RemoveAt(i);
+                    removed = true;
+                    break;
+                }
+            }
+            if (!removed)
+                throw new ScenarioException(
+                    $"No active Randomizer of type {randomizerType.Name} could be removed");
+        }
+
+        /// <summary>
+        /// Returns the execution order index of a randomizer of the given type
+        /// </summary>
+        /// <typeparam name="T">The type of randomizer to index</typeparam>
+        /// <returns>The randomizer index</returns>
+        /// <exception cref="ScenarioException"></exception>
+        public int GetRandomizerIndex<T>() where T : Randomizer, new()
+        {
+            for (var i = 0; i < m_Randomizers.Count; i++)
+            {
+                var randomizer = m_Randomizers[i];
+                if (randomizer is T)
+                    return i;
+            }
+            throw new ScenarioException($"A Randomizer of type {typeof(T).Name} was not added to this scenario");
+        }
+
+        /// <summary>
+        /// Moves a randomizer from one index to another
+        /// </summary>
+        /// <param name="currentIndex">The index of the randomizer to move</param>
+        /// <param name="nextIndex">The index to move the randomizer to</param>
+        public void ReorderRandomizer(int currentIndex, int nextIndex)
```

Comment:
Ok, I see that we are trying to maintain order via a list. There are ways around this list a SortedList, or, if the list never gets too big, I guess keeping it at a list is fine